src/app/page.tsx

```diff
@@ -1,116 +1,121 @@
-	const displayName = isERC1155Query.data
-		? nftQuery.data?.metadata.name
-		: contractMetadataQuery.data?.name;
+const chain = defineChain(defaultChainId);
+const contract = getContract({ address: defaultNftContractAddress, chain, client });
+const tokenId = defaultTokenId;
 
-	const description = isERC1155Query.data
-		? nftQuery.data?.metadata.description
-		: contractMetadataQuery.data?.description;
+async function getERCType() {
+    if (await isERC1155({ contract })) return "ERC1155";
+    if (await isERC721({ contract })) return "ERC721";
+    return "ERC20";
+}
 
-	const priceInWei = isERC1155Query.data
-		? claimCondition1155.data?.pricePerToken
-		: isERC721Query.data
-			? claimCondition721.data?.pricePerToken
-			: claimCondition20.data?.pricePerToken;
+async function getERC20Info() {
+    const claimCondition = await getActiveClaimCondition20({ contract });
+    return { claimCondition };
+}
 
-	const currency = isERC1155Query.data
-		? claimCondition1155.data?.currency
-		: isERC721Query.data
-			? claimCondition721.data?.currency
-			: claimCondition20.data?.currency;
+async function getERC721Info() {
+    const [claimCondition, metadata] = await Promise.all([
+        getActiveClaimCondition721({ contract }),
+        getContractMetadata({ contract }),
+    ]);
+    const priceInWei = claimCondition?.pricePerToken;
+    const currency = claimCondition?.currency;
+    const currencyMetadata = currency
+        ? await getCurrencyMetadata({ contract: getContract({ address: currency, chain, client }) })
+        : undefined;
+    return {
+        displayName: metadata?.data?.name,
+        description: metadata?.data?.description,
+        pricePerToken: currencyMetadata && priceInWei ? Number(toTokens(priceInWei, currencyMetadata.decimals)) : null,
+        currencyMetadata,
+        metadata,
+        claimCondition,
+    };
+}
 
-	const currencyContract = getContract({
-		address: currency || "",
-		chain,
-		client,
-	});
+async function getERC1155Info() {
+    const [nft, claimCondition] = await Promise.all([
+        getNFT({ contract, tokenId }),
+        getActiveClaimCondition1155({ contract, tokenId }),
+    ]);
+    const priceInWei = claimCondition?.pricePerToken;
+    const currency = claimCondition?.currency;
+    const currencyMetadata = currency
+        ? await getCurrencyMetadata({ contract: getContract({ address: currency, chain, client }) })
+        : undefined;
+    return {
+        displayName: nft?.metadata.name,
+        description: nft?.metadata.description,
+        pricePerToken: currencyMetadata && priceInWei ? Number(toTokens(priceInWei, currencyMetadata.decimals)) : null,
+        currencyMetadata,
+        nft,
+        claimCondition,
+    };
+}
 
-	const currencyMetadata = useReadContract(getCurrencyMetadata, {
-		contract: currencyContract,
-		queryOptions: { enabled: !!currency },
-	});
+type ERCInfo = {
+    displayName?: string;
+    description?: string;
+    pricePerToken?: number | null;
```

Comment:
are all these actually optional?

the UI must be pretty broken if none of these are set no?